src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/AuthenticatedUserIdTelemetryInitializer.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers
+{
+    using Microsoft.ApplicationInsights.AspNetCore.Extensions;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// This initializer sets the Authenticated User Id if enabled via <see cref="ApplicationInsightsServiceOptions.EnableAuthenticationTracking"/>.
+    /// </summary>
+    public class AuthenticatedUserIdTelemetryInitializer : TelemetryInitializerBase
+    {
+        private readonly bool enabled;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticatedUserIdTelemetryInitializer" /> class.
+        /// </summary>
+        /// <param name="options">Provides the option <see cref="ApplicationInsightsServiceOptions.EnableAuthenticationTracking"/> indicating whether the currently authenticated user is added to telemetries or not.</param>
+        /// <param name="httpContextAccessor">Accessor to provide HttpContext corresponding to telemetry items.</param>
+        public AuthenticatedUserIdTelemetryInitializer(IOptions<ApplicationInsightsServiceOptions> options, IHttpContextAccessor httpContextAccessor)
+             : base(httpContextAccessor)
+        {
+            this.enabled = options.Value.EnableAuthenticationTracking;
+        }
+
+        /// <inheritdoc />
+        protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
+        {
+            if (this.enabled)
+            {
+                if (string.IsNullOrEmpty(telemetry.Context.User.AuthenticatedUserId))
+                {
+                    var userIdentity = platformContext.User?.Identity;
+
+                    if (userIdentity != null &&
+                        userIdentity.IsAuthenticated)                    
+                    {
+                        telemetry.Context.User.AuthenticatedUserId = userIdentity.Name;
```

Comment:
WebUserTelemetryInitializer populates the id to the RequestTelemetry itself, so all telemetry associated within the scope of this Request will get it quickly.
https://github.com/microsoft/ApplicationInsights-aspnetcore/blob/develop/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/WebUserTelemetryInitializer.cs#L35

Did you intentionally not do it? 
I'll need to get someone more experienced that me to review this section - @SergeyKanzhelev @lmolkova @Dmitry-Matveev can one of you also to take a look?

Reply:
I agree. We should follow the pattern from `WebUserTelemetryInitializer` and `WebSessionTelemetryInitializer` and populate `AuthenticatedUserId` on `RequestTelemetry` - if not set already - and then populate `ITelemetry` from `RequestTelemetry`.